src/storage/src/file_purger.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_telemetry::debug;
+use store_api::storage::RegionId;
+use tokio::sync::Notify;
+
+use crate::scheduler::rate_limit::{BoxedRateLimitToken, RateLimitToken};
+use crate::scheduler::{Handler, LocalScheduler, Request};
+use crate::sst::AccessLayerRef;
+
+pub struct FilePurgeRequest {
+    pub region_id: RegionId,
+    pub file_name: String,
+    pub sst_layer: AccessLayerRef,
+}
+
+impl Request for FilePurgeRequest {
+    type Key = String;
+
+    fn key(&self) -> Self::Key {
+        format!("{}/{}", self.region_id, self.file_name)
+    }
+}
+
+pub struct FilePurgeHandler;
+
+#[async_trait::async_trait]
+impl Handler for FilePurgeHandler {
+    type Request = FilePurgeRequest;
+
+    async fn handle_request(
+        &self,
+        req: Self::Request,
+        token: BoxedRateLimitToken,
+        finish_notifier: Arc<Notify>,
+    ) -> crate::error::Result<()> {
+        req.sst_layer.delete_sst(&req.file_name).await?;
```

Comment:
Add a log if delete_sst returns error?
```suggestion
        req.sst_layer.delete_sst(&req.file_name).await.map_err(|e| error!(); e)?;
```